### PR TITLE
[netatmo] Correction of Last-Event group

### PR DESCRIPTION
**Pull request by clinique**
Complete resolution of issue https://github.com/openhab/openhab-addons/issues/15883 .

Enhanced:
- vpnUrl that should never be null.

### CODE REVIEW

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/EventChannelHelper.java [hunk 1]
```diff
@@ -66,45 +67,39 @@ public void setNewData(@Nullable NAObject data) {
...
+            case CHANNEL_EVENT_VIDEO_STATUS ->
+                event.getVideoId() != null ? toStringType(event.getVideoStatus()) : UnDefType.NULL;
+            case CHANNEL_EVENT_VIDEO_LOCAL_URL -> getStreamURL(true, event.getVideoId(), event.getVideoStatus());
+            case CHANNEL_EVENT_VIDEO_VPN_URL -> getStreamURL(false, event.getVideoId(), event.getVideoStatus());
+            default -> null;
+        };
+    }
+
+    private String getUrl(boolean local) {
+        return Objects.requireNonNull(local ? localUrl : vpnUrl);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/EventChannelHelper.java:96]: Same question as in the other PR: how can you be sure that `localUrl` is not null ?

**Comment by lolodomo** [comment]
Conflict has to be resolved after the merge of another PR.

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/EventChannelHelper.java [hunk 2]
```diff
@@ -66,45 +66,39 @@ public void setNewData(@Nullable NAObject data) {
...
     }
 
     private State getStreamURL(boolean local, @Nullable String videoId, VideoStatus videoStatus) {
-        String url = local ? localUrl : vpnUrl;
-        if ((local && !isLocal) || url == null || videoId == null || videoStatus != VideoStatus.AVAILABLE) {
+        if ((local && localUrl == null) || videoId == null || videoStatus != VideoStatus.AVAILABLE) {
             return UnDefType.NULL;
         }
-        return toStringType("%s/vod/%s/index.m3u8", url, videoId);
+        return toStringType("%s/vod/%s/index.m3u8", getUrl(local), videoId);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/EventChannelHelper.java:102]: Please keep a check that `getUrl(local)` is not null, something like:
```
String url = getUrl(local);
if (url == null || videoId == null || videoStatus != VideoStatus.AVAILABLE) {
    return UnDefType.NULL;
}
return toStringType("%s/vod/%s/index.m3u8", url, videoId);
```

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/EventChannelHelper.java [hunk 3]
```diff
@@ -36,8 +37,8 @@
  */
 @NonNullByDefault
 public class EventChannelHelper extends ChannelHelper {
-    private boolean isLocal;
-    private @Nullable String vpnUrl, localUrl;
+    private @NonNullByDefault({}) String vpnUrl;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/EventChannelHelper.java:40]: Please avoid bypassing the null checks. I would prefer to keep a `@Nullable` for `vpnUrl`.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/EventChannelHelper.java (hunk 3 above)
Comment (clinique) [reply bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/EventChannelHelper.java:40]: Done

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**